OpenRA.Game/Renderer.cs

```diff
@@ -123,26 +128,49 @@ public void InitializeDepthBuffer(MapGrid mapGrid)
 		public void BeginFrame(int2 scroll, float zoom)
 		{
 			Context.Clear();
+
+			// Cache property results to avoid the unnecessary overhead of repeat queries
+			var resolution = Resolution;
+			var scale = WindowScale;
+
+			var backingResolution = new Size((int)(scale * resolution.Width), (int)(scale * resolution.Height));
+			var backingBufferSize = backingResolution.NextPowerOf2();
+			if (screenSprite == null || backingBufferSize != screenSprite.Sheet.Size)
```

Comment:
This needs to check for `screenSprite.Bounds.Width != backingBufferSize.Width`.

`backingResolution` and `backingBufferSize` should also be renamed to `SurfaceSize` to match the terminology inside `Sdl2PlatformWindow`.